### PR TITLE
fix: ready to use cli with create-clip command

### PR DESCRIPTION
**Pull request by prieston**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
 const utils = require("./utils.js");
 
-const command = yargs.argv._[0];
-const projectName = yargs.argv._[1];
+// const command = yargs.argv._[0];
+// const projectName = yargs.argv._[1];
+
+// if (!utils.checkCommand(command)) return utils.showHelp();
+// if (!utils.checkName(projectName)) return utils.showHelp();
+// utils.cloneGitRepo(command, projectName);
```
Comment (alexdor) [inline src/index.js:11]: Do we need these here?

File: src/utils.js [hunk 2]
```diff
@@ -1,82 +1,112 @@
...
+const { MESSAGES, STARTER_REPO } = require("./config");
 const fs = require("fs");
-const rimraf = require("rimraf");
 const process = require("process");
-const { exec } = require("child_process");
+const util = require("util");
+const exec = util.promisify(require("child_process").exec);
+const rimraf = util.promisify(require("rimraf"));
+const copyfiles = util.promisify(require("copyfiles"));
+// const copyfiles = require("copyfiles");
```
Comment (alexdor) [inline src/utils.js:9]:
```suggestion
```

File: src/utils.js [hunk 3]
```diff
@@ -1,82 +1,112 @@
...
-  return !!name;
-}
-
-function showHelp() {
-  console.log(USAGE);
-}
+async function cloneGitRepo(commandName, projectName, options) {
+  // const clipDest = path.join("./", projectName, "clip");
+  // const clipFrom = path.join("./", projectName, "dist", "clip", "**", "*");
+  // await copyfiles([clipFrom, clipDest], { up: 3 });
```
Comment (alexdor) [inline src/utils.js:19]:
```suggestion
```

File: src/utils.js [hunk 4]
```diff
@@ -1,82 +1,112 @@
...
+    if (options.js) {
+      throbber = ora(MESSAGES.javascript).start();
+
+      const clipDest = path.join("./", projectName, "clip");
+      const clipFrom = path.join("./", projectName, "dist", "clip", "**", "*");
+      await copyfiles([clipFrom, clipDest], { up: 3 });
+
+      const serverDest = path.join("./", projectName, "server");
+      const serverFrom = path.join(
+        "./",
```
Comment (alexdor) [inline src/utils.js:72]: Shouldn't this be the same?
```suggestion
        ".",
```

File: src/utils.js [hunk 5]
```diff
@@ -1,82 +1,112 @@
...
+        projectName,
+        "dist",
+        "server",
+        "**",
+        "*"
+      );
+      await copyfiles([serverFrom, serverDest], { up: 3 });
+
+      const distPath = path.join("./", projectName, "dist");
+      const tsconfig = path.join("./", projectName, "tsconfig.json");
```
Comment (alexdor) [inline src/utils.js:82]: The ./ works only for unix machines so this will break on windows

File: src/utils.js [hunk 6]
```diff
@@ -1,82 +1,112 @@
...
+      await rimraf(distPath);
+      await rimraf(tsconfig);
+      await rimraf("./**/*.ts");
```
Comment (alexdor) [inline src/utils.js:85]: The ./ works only for unix machines so this will break on windows

**Review by alexdor — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (prieston) [reply src/index.js:11]: nope

**Review by alexdor — COMMENTED** [review]
Don’t we also need to update the docs?

also, I’ve seen the command “npm create” been mentioned in a lot of packages (when initializing something) should we support it?

**Review by alexdor — APPROVED** [review]

**Pull request merged**